### PR TITLE
[SPARK-28202] [Core] [Test] Avoid noises of system props in SparkConfSuite

### PR DESCRIPTION
**Pull request by LiShuMing**
When SPARK_HOME of env is set and contains a specific `spark-defaults,conf`, `org.apache.spark.util.loadDefaultSparkProperties` method may noise `system props`. So when runs `core/test` module, it is possible to fail to run `SparkConfSuite` .

 It's easy to repair by setting `loadDefaults` in `SparkConf` to be false.

```
[info] - deprecated configs *** FAILED *** (79 milliseconds)
[info] 7 did not equal 4 (SparkConfSuite.scala:266)
[info] org.scalatest.exceptions.TestFailedException:
[info] at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:528)
[info] at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:527)
[info] at org.scalatest.FunSuite.newAssertionFailedException(FunSuite.scala:1560)
[info] at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:501)
[info] at org.apache.spark.SparkConfSuite.$anonfun$new$26(SparkConfSuite.scala:266)
[info] at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info] at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info] at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info] at org.scalatest.Transformer.apply(Transformer.scala:22)
[info] at org.scalatest.Transformer.apply(Transformer.scala:20)
[info] at org.scalatest.FunSuiteLike$$anon$1.apply(FunSuiteLike.scala:186)
[info] at org.apache.spark.SparkFunSuite.withFixture(SparkFunSuite.scala:149)
[info] at org.scalatest.FunSuiteLike.invokeWithFixture$1(FunSuiteLike.scala:184)
[info] at org.scalatest.FunSuiteLike.$anonfun$runTest$1(FunSuiteLike.scala:196)
[info] at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
```

### CODE REVIEW

**Comment by jerryshao** [comment]
ok to test.

**Comment by jerryshao** [comment]
Jenkins, test this please.

**Comment by srowen** [comment]
This test doesn't fail in a normal test run -- why would you customize the test env? that seems like the problem?

**Comment by LiShuMing** [comment]
Thanks for @srowen 's replies. However I think's it's a better to prevent the cases in coeds, and it seems more reasonable?

**Comment by srowen** [comment]
This change is probably OK anyway, but, would you not have other problems if you're locally modifying system properties during tests?

**Comment by jerryshao** [comment]
I would guess that other tests will also be failed possibly if you have local modified properties. Have you met this issue in other test suites?

**Comment by LiShuMing** [comment]
In `core/test`  module, only this unittest failed. Maybe I need check other modules?

**Comment by jerryshao** [comment]
Jenkins, test this please.

**Comment by jerryshao** [comment]
I'm OK with the changes.

File: core/src/test/scala/org/apache/spark/SparkConfSuite.scala [hunk 1]
```diff
@@ -242,7 +242,8 @@ class SparkConfSuite extends SparkFunSuite with LocalSparkContext with ResetSyst
   }
 
   test("deprecated configs") {
-    val conf = new SparkConf()
+    // Set `loadDefaults` to be false, so avoid system props's noises.
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/SparkConfSuite.scala]: I'd remove this comment, I'm not sure what the 'noise' is here. It's really a workaround for your env.

Comment (LiShuMing) [reply core/src/test/scala/org/apache/spark/SparkConfSuite.scala]: Thanks @srowen 's replies. I resolved the comments.

**Review by srowen — APPROVED** [review]

**Comment by jerryshao** [comment]
Thanks @LiShuMing , merging to master.

**Pull request closed without merging**